In [3]:
!pip install -q gdown
!gdown --id 1iO8PfMaaO1raQ4VO9aL8T5YT52NBXWLC -O inegi_raw.csv
!gdown --id 1lfeNvdnveMZ3W8NPpe2DYi8pKMsOGut0 -O coneval_raw.csv

import pandas as pd

inegi_raw = pd.read_csv("inegi_raw.csv", encoding="latin1")
coneval_raw = pd.read_csv("coneval_raw.csv", encoding="latin1")

/usr/local/lib/python3.11/dist-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  warnings.warn(
Downloading...
From (original): https://drive.google.com/uc?id=1iO8PfMaaO1raQ4VO9aL8T5YT52NBXWLC
From (redirected): https://drive.google.com/uc?id=1iO8PfMaaO1raQ4VO9aL8T5YT52NBXWLC&confirm=t&uuid=9fce4a34-053b-457a-a6bb-22e1da78070d
To: /content/inegi_raw.csv
100% 150M/150M [00:01<00:00, 130MB/s]
/usr/local/lib/python3.11/dist-packages/gdown/__main__.py:140: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  warnings.warn(
Downloading...
From: https://drive.google.com/uc?id=1lfeNvdnveMZ3W8NPpe2DYi8pKMsOGut0
To: /content/coneval_raw.csv
100% 582k/582k [00:00<00:00, 77.0MB/s]


<ipython-input-3-0e19e927ea94>:7: DtypeWarning: Columns (8) have mixed types. Specify dtype option on import or set low_memory=False.
  inegi_raw = pd.read_csv("inegi_raw.csv", encoding="latin1")


In [5]:
inegi_raw.head()


ENTIDAD         NOM_ENT  MUN                             NOM_MUN   LOC  \
0           0  Total nacional    0                      Total nacional     0   
1           0  Total nacional    0                      Total nacional  9998   
2           0  Total nacional    0                      Total nacional  9999   
3           1  Aguascalientes    0  Total de la entidad Aguascalientes     0   
4           1  Aguascalientes    0  Total de la entidad Aguascalientes  9998   

                        NOM_LOC LONGITUD LATITUD ALTITUD     POBTOT  ...  \
0                Total nacional      NaN     NaN     NaN  126014024  ...   
1   Localidades de una vivienda      NaN     NaN     NaN     250354  ...   
2  Localidades de dos viviendas      NaN     NaN     NaN     147125  ...   
3           Total de la Entidad      NaN     NaN     NaN    1425607  ...   
4   Localidades de una vivienda      NaN     NaN     NaN       3697  ...   

    VPH_CEL VPH_INTER  VPH_STVP VPH_SPMVPI  VPH_CVJ VPH_SINRTV VPH_SINLTC  \
0  30775898  18307193  15211306    6616141  4047100    1788552    3170894   
1     47005      8385     18981       1732     1113      12775      14143   
2     25581      5027     11306        971      708       8247      10065   
3    359895    236003    174089      98724    70126       6021      15323   
4       732       205       212         48       41         39         62   

  VPH_SINCINT VPH_SINTIC TAMLOC  
0    15108204     852871      *  
1       51293       7154      *  
2       29741       5283      *  
3      128996       1711      *  
4         530         20      *  

[5 rows x 286 columns]

In [6]:
coneval_raw.head()

clave_entidad entidad_federativa  clave_municipio       municipio  \
0              1     Aguascalientes             1001  Aguascalientes   
1              1     Aguascalientes             1002        Asientos   
2              1     Aguascalientes             1003        Calvillo   
3              1     Aguascalientes             1004           Cosío   
4              1     Aguascalientes             1005     Jesús María   

  poblacion pobreza pobreza_pob pobreza_e pobreza_e_pob pobreza_m  ... ic_ali  \
0   922,268    23.7     218,414       2.0        18,206      21.7  ...   15.8   
1    48,635    40.1      19,518       4.1         2,015      36.0  ...   28.5   
2    52,377    45.8      23,966       4.5         2,356      41.3  ...   20.1   
3    15,942    37.0       5,905       3.4           538      33.7  ...   25.0   
4   127,962    26.3      33,708       3.3         4,204      23.1  ...   17.9   

  ic_ali_pob carencias carencias_pob carencias3 carencias3_pob   plp  plp_pob  \
0    145,444      51.6       475,558        7.2         66,231  33.7  310,444   
1     13,870      79.7        38,761       15.7          7,650  46.8   22,737   
2     10,552      86.2        45,164       14.4          7,520  49.5   25,935   
3      3,980      69.3        11,043       10.9          1,736  47.0    7,488   
4     22,945      59.3        75,894       11.0         14,069  34.0   43,565   

  plp_e plp_e_pob  
0   9.0    82,866  
1  15.8     7,683  
2  16.0     8,380  
3  16.5     2,634  
4  10.9    13,912  

[5 rows x 37 columns]

In [18]:
#Nos quedamos únicamente con el conjunto de columnas del coneval que queremos: 'clave_entidad', 'municipio','ic_sbv_pob', 'pobreza_pob'
coneval_raw = coneval_raw[['clave_entidad', 'municipio','ic_sbv_pob', 'pobreza_pob']]


In [19]:
#Nos quedamos únicamente con el total de cada municipio (ignoramos localidades)
inegi_raw = inegi_raw[inegi_raw["NOM_LOC"].str.strip() == "Total del Municipio"].copy()
#Para hacer el merge necesitamos renombrar algunas columnas
inegi_raw.rename(columns={'NOM_MUN': 'municipio', 'ENTIDAD': 'clave_entidad'}, inplace=True)

In [20]:
inegi_raw.head()

clave_entidad         NOM_ENT  MUN       municipio  LOC  \
6                1  Aguascalientes    1  Aguascalientes    0   
574              1  Aguascalientes    2        Asientos    0   
749              1  Aguascalientes    3        Calvillo    0   
918              1  Aguascalientes    4          Cosío    0   
985              1  Aguascalientes    5   Jesús María    0   

                 NOM_LOC LONGITUD LATITUD ALTITUD  POBTOT  ... VPH_CEL  \
6    Total del Municipio      NaN     NaN     NaN  948990  ...  251719   
574  Total del Municipio      NaN     NaN     NaN   51536  ...   10682   
749  Total del Municipio      NaN     NaN     NaN   58250  ...   13666   
918  Total del Municipio      NaN     NaN     NaN   17000  ...    3424   
985  Total del Municipio      NaN     NaN     NaN  129929  ...   31408   

    VPH_INTER VPH_STVP VPH_SPMVPI VPH_CVJ VPH_SINRTV VPH_SINLTC VPH_SINCINT  \
6      178619   130290      80951   56131       3299       7293       74227   
574      4526     3882        596     556        377       1447        7206   
749      6553     4749       1382    1352        443        904        8061   
918      1741     1664        223     191        139        414        1912   
985     19920    13483       9296    6582        480       1150       11447   

    VPH_SINTIC TAMLOC  
6          731      *  
574        177      *  
749        138      *  
918         50      *  
985        148      *  

[5 rows x 286 columns]

In [21]:
#hacemos el merge
df_raw = pd.merge(inegi_raw,coneval_raw, on=['municipio','clave_entidad'], how='left')

In [23]:
df_raw.head()

clave_entidad         NOM_ENT  MUN       municipio  LOC  \
0              1  Aguascalientes    1  Aguascalientes    0   
1              1  Aguascalientes    2        Asientos    0   
2              1  Aguascalientes    3        Calvillo    0   
3              1  Aguascalientes    4          Cosío    0   
4              1  Aguascalientes    5   Jesús María    0   

               NOM_LOC LONGITUD LATITUD ALTITUD  POBTOT  ... VPH_STVP  \
0  Total del Municipio      NaN     NaN     NaN  948990  ...   130290   
1  Total del Municipio      NaN     NaN     NaN   51536  ...     3882   
2  Total del Municipio      NaN     NaN     NaN   58250  ...     4749   
3  Total del Municipio      NaN     NaN     NaN   17000  ...     1664   
4  Total del Municipio      NaN     NaN     NaN  129929  ...    13483   

  VPH_SPMVPI VPH_CVJ VPH_SINRTV VPH_SINLTC VPH_SINCINT VPH_SINTIC TAMLOC  \
0      80951   56131       3299       7293       74227        731      *   
1        596     556        377       1447        7206        177      *   
2       1382    1352        443        904        8061        138      *   
3        223     191        139        414        1912         50      *   
4       9296    6582        480       1150       11447        148      *   

  ic_sbv_pob pobreza_pob  
0     10,473     218,414  
1      3,604      19,518  
2      1,612      23,966  
3        NaN         NaN  
4        NaN         NaN  

[5 rows x 288 columns]

In [22]:
df_raw.shape

(2469, 288)

El dataset integrado es consistente, tenemos 2469 observaciones y 288 columnas, dos más que las que teníamos en el dataset original del INEGI

In [24]:
#guardamos el dataset como Equipo7-DatosIntegrados.csv
df_raw.to_csv("Equipo7-DatosIntegrados.csv", index=False)